HW1/First lesson.playground/Contents.swift

```diff
@@ -0,0 +1,217 @@
+import UIKit
+
+//first task
+protocol HomeworkService {
+    // Функция деления с остатком, должна вернуть в первой части результат деления, во второй части остаток.
+    func divideWithRemainder(_ x: Int, by y: Int) -> (Int, Int)
+
+    // Функция должна вернуть числа фибоначчи.
+    func fibonacci(n: Int) -> [Int]
+
+    // Функция должна выполнить сортировку пузырьком.
+    func sort(rawArray: [Int]) -> [Int]
+
+    // Функция должна преобразовать массив строк в массив первых символов строки.
+    func firstLetter(strings: [String]) -> [Character]
+
+    // Функция должна отфильтровать массив по условию, которое приходит в параметре `condition`. (Нельзя юзать `filter` у `Array`)
+    func filter(array: [Int], condition: ((Int) -> Bool)) -> [Int]
+}
+
+struct Service: HomeworkService{
+    func divideWithRemainder(_ x: Int, by y: Int) -> (Int, Int) {
+        let first = x/y
+        let second = x%y
+        return(first, second)
+    }
+    
+    func fibonacci(n: Int) -> [Int] {
+        var result = [0, 1]
+        
+        guard n > 1 else {
+            return result
+        }
+        
+        while result.count < n {
+            let nextNumber = result[result.count - 1] + result[result.count - 2]
+            result.append(nextNumber)
+        }
+        
+        return result
+    }
+    
+    func sort(rawArray: [Int]) -> [Int] {
+        var array = rawArray
+        
+        for i in 0..<array.count {
+            for j in 0..<(array.count - i - 1) {
+                if array[j] > array[j + 1] {
+                    let temp = array[j]
+                    array[j] = array[j + 1]
+                    array[j + 1] = temp
+                }
+            }
+        }
+        
+        return array
+    }
+    
+    func firstLetter(strings: [String]) -> [Character] {
+        var result = [Character]()
+        
+        for string in strings {
+            if let firstChar = string.first {
+                result.append(firstChar)
+            }
+        }
+        
+        return result
+    }
+    
+    func filter(array: [Int], condition: ((Int) -> Bool)) -> [Int] {
+        var result = [Int]()
+        
+        for element in array {
+            if condition(element) {
+                result.append(element)
+            }
+        }
+        
+        return result
+    }
+}
+
+
+//second task
+// Книги, которые могут быть выданы в библиотеке
+protocol Borrowable {
+    var isBorrowed: Bool { get set }
+    func borrow()
+    func returnBook()
+}
+
+// Типы книг
+enum BookType {
+    case fiction
+    case nonFiction
+    case reference
+}
+
+// Класс книги
+class Book: Borrowable {
+    var title: String
+    var author: String
+    var isBorrowed: Bool = false
+    
+    init(title: String, author: String) {
+        self.title = title
+        self.author = author
+    }
+    
+    func borrow() {
+        if !isBorrowed {
```

Comment:
guard лучше, меньше вложенности
```suggestion
        guard !isBorrowed else {
            print("Книга '\(title)' уже взята в библиотеке.")
            return
        }
        
        isBorrowed = true
        print("Книга '\(title)' была взята в библиотеке.")
```